miniqiita.go

```diff
@@ -121,12 +115,7 @@ func (c *Client) doRequest(req *http.Request, respBody interface{}) (int, error)
 		return resp.StatusCode, nil
 	}
 
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return 0, err
-	}
-
-	if err := json.Unmarshal(bodyBytes, respBody); err != nil {
+	if err := json.NewDecoder(resp.Body).Decode(respBody); err != nil {
```

Comment:
`json.Decoder` is better than `ioutil.ReadAll` from the viewpoint of the memory usage and performance.
ref: https://www.datadoghq.com/blog/crossing-streams-love-letter-gos-io-reader/

Reply:
📝